Work on app.R to expose prior therapy

jcallaway opened the pull request:
Please DO NOT MERGE to master yet, as this PR is only intended as a vehicle to ask questions. This small change just queries the new DB table for prior therapy info, and blindly displays it next to the corresponding trial, regardless of whether any of the associated thesaurus terms match the patient's prior therapies.  I'm submitting it for review at this stage since I suspect that I might be missing something, even at this simple level.

There are TODOs in the code showing where I need to call the new check_if_any() function that will implement the logic we discussed in the last meeting.

Commit:
Work on app.R to expose prior therapy

Please DO NOT MERGE to master yet, as this PR is only intended as a vehicle to ask questions.
This small change just queries the new DB table for prior therapy info, and blindly displays it next to the corresponding trial, regardless of whether any of the associated thesaurus terms match the patient's prior therapies.  I'm submitting it for review at this stage since I suspect that I might be missing something, even at this simple level.

There are TODOs in the code showing where I need to call the new check_if_any() function that will implement the logic we discussed in the last meeting.

huberthickman commented on app.R:
```diff
@@ -1830,6 +1830,19 @@ select count(nct_id) as number_sites, nct_id from trial_sites where org_status =
+      "select nct_id, nci_thesaurus_concept_id from trial_prior_therapies where eligibility_criterion='inclusion'")
+    trials_with_prior_therapy_exc_df <- safe_query(dbGetQuery,
+      "select nct_id, nci_thesaurus_concept_id from trial_prior_therapies where eligibility_criterion='exclusion'")
+
```
This two SQL statements should also add 

`inclusion_indicator = 'TRIAL' `  to their WHERE clauses.  The 'TREE" ones are just rollups for that trial (i.e. ancestors going up the DAG)

huberthickman commented on app.R:
```diff
@@ -1830,6 +1830,19 @@ select count(nct_id) as number_sites, nct_id from trial_sites where org_status =
+    # any inclusion/exclusion as I am doing here.
+    df_crit <- transform(df_crit, pt_inc_matches = ifelse(clean_nct_id %in% trials_with_prior_therapy_inc_df$nct_id, TRUE, FALSE))
+    df_crit <- transform(df_crit, pt_exc_matches = ifelse(clean_nct_id %in% trials_with_prior_therapy_exc_df$nct_id, TRUE, FALSE))
+
```
These need to be functional applys -- and perhaps with a function that returns three values  0 for false, 1 for true, and .5 for the maybes in the inclusion sense.  

I..e check_if_any(first TRIAL code in prior therapy inclusion) || check if any (second TRIAL code in pt inclusion || ... || ) -- if that is TRUE, then it would return TRUE.  
ELSE (i.e. none of those return TRUE) 
     -- Turn the sense of the checking around....  Have an environment of the criteria codes, and call the check_if_anyon the stuff in patient data environment NOTE -- can skip this for the moment to get the simple case working.
ELSE FALSE

Commit:
Call check_if_any() with the prior therapy codes entered by the user.

TODO: implement new logic for prior therapy matching; this will require separate functions for inclusion or exclusion.

Commit:
Add new exclusion matching logic.

Commit:
Clarify comments.
